cmake/app/boilerplate.cmake

```diff
@@ -217,8 +217,25 @@ list(APPEND ARCH_ROOT ${ZEPHYR_BASE})
 # Check that BOARD has been provided, and that it has not changed.
 zephyr_check_cache(BOARD REQUIRED)
 
+# Search for revision and layour serparators
 string(FIND "${BOARD}" "@" REVISION_SEPARATOR_INDEX)
-if(NOT (REVISION_SEPARATOR_INDEX EQUAL -1))
+string(FIND "${BOARD}" "%" LAYOUT_SEPARATOR_INDEX)
+
+if((NOT (REVISION_SEPARATOR_INDEX EQUAL -1)) AND (NOT (REVISION_SEPARATOR_INDEX EQUAL -1)))
```

Comment:
The second condition probably needs to check `LAYOUT_SEPARATOR_INDEX`

Reply:
Fixed, thanks